cds-snc/cppd-medical-report-sails
# Fixing dockerfile and adding docker ignore

## Description
Adding Docker Ignore so we don't send node_modules to the context since we reinstall all modules when building 

Modifying the docker file so that dependencies are separate from the rest of the app to speed up the dev/test/build loop for infra work.

## To Test

1. Build Container
2. Run Container
3. Check if the site comes up

## Not done in PR

- Multistage builds
- `.dockerignore` isn't as fine tuned as it could be should only pass what it needs to build to context
- Production Builds are not occurring we will need database schema's for that since Prod forces the database alteration flag to be safe. 
- Probably some other things as well.
